internal/cli/main.go

```diff
@@ -67,10 +67,20 @@ func printUsageAndExit() {
 
 func dbMakeConnectionString(driver, user, password, address, name, ssl string) string {
 	return fmt.Sprintf("%s://%s:%s@%s/%s?sslmode=%s",
```

Comment:
Consider using [`URL.String()`](https://pkg.go.dev/net/url#URL.String) to assemble this string. It will apply the required escaping. [`url.UserPassword()`](https://pkg.go.dev/net/url#UserPassword) will encapsulate the username and password.